### PR TITLE
fix(autocomplete): do not trigger submit on ENTER

### LINKED ISSUE #3159: md-autocomplete trigger submit on ENTER
md-autocomplete trigger submit form by ENTER click on option:
Plunker template: https://plnkr.co/edit/0a8lqrodE9WVTZEJprlI?p=preview

### PR DESCRIPTION
**Pull request by jefersonestevo**
When an option is selected with the ENTER key, it should not submit the form and prevent the default input behavior

Fixes #3159

### CODE REVIEW

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 1]
```diff
@@ -611,6 +611,24 @@ describe('MdAutocomplete', () => {
       });
     }));
 
+    it('should not call the default input behaviour when an option is selected with ENTER',
+        async(() => {
+      fixture.whenStable().then(() => {
```
Comment (crisbeto) [inline src/lib/autocomplete/autocomplete.spec.ts]: I don't think you need the `fixture.whenStable` here.

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 2]
```diff
@@ -611,6 +611,24 @@ describe('MdAutocomplete', () => {
...
+        fixture.componentInstance.trigger._handleKeydown(DOWN_ARROW_EVENT);
+
+        fixture.whenStable().then(() => {
+          fixture.detectChanges();
```
Comment (crisbeto) [inline src/lib/autocomplete/autocomplete.spec.ts]: You shouldn't need to call `detectChanges` here and below on line 625.

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 3]
```diff
@@ -611,6 +611,24 @@ describe('MdAutocomplete', () => {
       });
     }));
 
+    it('should not call the default input behaviour when an option is selected with ENTER',
```
Comment (crisbeto) [inline src/lib/autocomplete/autocomplete.spec.ts]: You could shorten this to something like `should prevent the default enter key action` so it doesn't have to be on separate lines.

File: src/lib/autocomplete/autocomplete.spec.ts (hunk 2 above)
Comment (jefersonestevo) [inline src/lib/autocomplete/autocomplete.spec.ts]: Done. I've removed the extra `detectChanges` but had to keep the `fixture.whenStable()` so that the test could simulate the "open" and "select first option" action before it press the enter key
